File: 141. Linked List Cycle/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
    if head == nil {
        return false
    }
    result := false
    cur := new(ListNode)
    cur = head
    for {
        if cur.Next == nil {
            break
        }
        if cur.Val == 10000 {
            result = true
            break
        }
        cur.Val = 10000
        cur = cur.Next
    }
    return result
}

File: 141. Linked List Cycle/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let cur = head
    let records = new Map()
    while(cur) {
        if(records.has(cur)) {
            return true
        }
        records.set(cur, 1)
        cur = cur.next
    }
    return false
};